src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java

```diff
@@ -0,0 +1,143 @@
+import java.net.http.HttpResponse;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class SettingsProvider implements ISettingsProvider {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Gson GSON = new GsonBuilder().registerTypeHierarchyAdapter(ResourceLocation.class, new ResourceLocationTypeAdapter()).create();
+
+    private final HttpClient client;
+    private final String baseUrl;
+    private final String apiVersion;
+    private final Map<String, String> settingValues = new HashMap<>();
+
+    public SettingsProvider(String baseUrl, String apiVersion) {
+        this.baseUrl = baseUrl;
+        this.apiVersion = apiVersion;
+        this.client = HttpClient.newHttpClient();
+    }
+
+    public void syncSettingsCache() {
+        getSettingValuesAsync().handleAsync(this::checkSettings).thenAccept(x -> {
+            this.settingValues.clear();
+            this.settingValues.putAll(x);
+        });
+    }
+
+    @Nullable
+    public String getSettingsValue(String key) {
+        return this.settingValues.get(key);
+    }
+
+    public @NotNull Optional<String> getSettingsValueOpt(String key) {
+        return Optional.ofNullable(this.settingValues.get(key));
+    }
+
+    public boolean isSettingTrue(String key) {
+        return "true".equals(this.settingValues.get(key));
+    }
+
+    @Override
+    public @NotNull CompletableFuture<Collection<Supporter>> getSupportersAsync() {
+        return GetSupportersAsync().handleAsync(this::checkSupporter);
+    }
+
+    @Nullable
+    public CompletableFuture<String> getSettingValueAsync(String key) {
+        return get("config/get?configId=" + key);
+    }
+
+    public CompletableFuture<Map<String, String>> getSettingValuesAsync() {
+        return get("config/list").thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(Map.class, String.class, String.class).getType()));
+    }
+
+    public CompletableFuture<Map<String, String>> getSettingValuesAsync(String modid) {
+        return get("config/list?modid=" + modid).thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(Map.class, String.class, String.class).getType()));
+    }
+
+    public CompletableFuture<Collection<Supporter>> GetSupportersAsync() {
+        Type type = TypeToken.getParameterized(List.class, Supporter.class).getType();
+        return get("supporter/list").thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(List.class, Supporter.class).getType()));
+    }
+
+    public CompletableFuture<Collection<Supporter>> GetSupportersAsync(String modid) {
+        return GetSupportersAsync().thenApply(x -> x.stream().filter(y -> y.faction().getNamespace().equals(modid)).collect(Collectors.toList()));
+    }
+
+    private CompletableFuture<String> get(String path) {
+        return get(this.apiVersion, path);
+    }
+
+    private CompletableFuture<String> get(String apiVersion, String path) {
```

Comment:
Why is api version separate from base url?
If we add a new api version on the server, the mod should completely switch to the new api, right?